components/vercel/vercel.app.mjs

```diff
@@ -1,11 +1,161 @@
+import { axios } from "@pipedream/platform";
+import constants from "./constants.mjs";
+
 export default {
   type: "app",
   app: "vercel",
-  propDefinitions: {},
+  propDefinitions: {
+    project: {
+      type: "string",
+      label: "Project",
+      description: "Filter deployments from the given projectId",
+      optional: true,
+      async options() {
+        const projects = await this.listProjects();
+        if (!projects || projects.length === 0) {
+          return [];
+        }
+        return projects.map((project) => ({
+          label: project.name,
+          value: project.id,
+        }));
+      },
+    },
+    deployment: {
+      type: "string",
+      label: "Deployment",
+      description: "Select the deployment to cancel",
+      async options({ state }) {
+        const params = state
+          ? {
+            state,
+          }
+          : undefined;
```

Comment:
If `state` is `undefined`, shouldn't `params = {}`? Since later in `paginate()` it spreads `config.params`?